File: app/src/main/java/com/example/rssreader/presentation/feed/FeedView.java
package com.example.rssreader.presentation.feed;

import android.support.annotation.NonNull;

import com.example.rssreader.data.RssItem;

import java.util.List;

public interface FeedView {
    void showData(@NonNull List<RssItem> data);
    void showError();
}

File: app/src/main/java/com/example/rssreader/presentation/feed/FeedView.kt
package com.example.rssreader.presentation.feed

import com.example.rssreader.domain.entity.RssItem

interface FeedView {
    fun showProgress()
    fun hideProgress()

    fun showData(data: List<RssItem>)
    fun showError()

    fun openDetails(url: String)
}
